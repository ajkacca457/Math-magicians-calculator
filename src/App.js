function App() {
  return (
    <div className="App">
      <h1>Math magicians calculator</h1>
    </div>
  );
}

export default App;
