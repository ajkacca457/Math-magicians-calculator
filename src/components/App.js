import React, { useState } from 'react';
import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';
// eslint-disable-next-line no-unused-vars
import calculate from '../logic/calculate';
import Home from './Home';
import Quote from './Quote';
import Navbar from './Navbar';
import Calculator from './Calculator';
import '@fortawesome/fontawesome-free/css/all.min.css';

const data = {
  total: null,
  next: null,
  operation: null,
};

const App = () => {
  const [state, setState] = useState(data);

  const handleClick = buttonName => {
    const newstate = calculate(state, buttonName);
    setState(newstate);
  };

  const updateResult = () => {
    // eslint-disable-next-line no-unused-vars
    const { total, next, operation } = state;
    if (next) {
      return next;
    }
    return total;
  };

  return (
    <Router>
      <div className="app">
        <Navbar />
        <Route exact path="/">
          <Redirect to="/home" />
        </Route>
        <Route exac path="/home" component={Home} />
        <Route
          exact
          path="/calculator"
          render={() => (
            <Calculator
              updateResult={updateResult}
              handleClick={handleClick}
            />
          )}
        />
        <Route exac path="/quote" component={Quote} />
      </div>
    </Router>
  );
};

export default App;
