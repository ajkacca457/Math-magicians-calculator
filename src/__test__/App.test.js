import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../components/App';

describe('App Component', () => {
  it('renders correctly', () => {
    const apptree = TestRenderer
      .create(<App />)
      .toJSON();
    expect(apptree).toMatchSnapshot();
  });
});
