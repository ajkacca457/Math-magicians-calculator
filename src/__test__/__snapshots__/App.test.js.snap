// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="navbar"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f5913e",
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "0px",
        "padding": "5px 10px",
      }
    }
  >
    <h3
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Math Magician Calculator
      <i
        className="fas fa-magic"
        style={
          Object {
            "marginLeft": "5px",
          }
        }
      />
    </h3>
    <div
      className="links"
    >
      <a
        className="link"
        href="/home"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "marginRight": "15px",
            "textDecoration": "none",
          }
        }
      >
        <i
          className="fas fa-home"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Home
      </a>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "marginRight": "15px",
            "textDecoration": "none",
          }
        }
      >
        <i
          className="fas fa-calculator"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Calculator
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "marginRight": "15px",
            "textDecoration": "none",
          }
        }
      >
        <i
          className="far fa-comments"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Quote
      </a>
    </div>
  </div>
  <div
    className="home"
    style={
      Object {
        "margin": "10% auto",
        "width": "80%",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "orange",
          "fontSize": "140%",
          "textAlign": "left",
          "textDecoration": "underline",
        }
      }
    >
      Welcome to our page!
    </h1>
    <p
      style={
        Object {
          "lineHeight": "1.8",
          "textAlign": "justify",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </p>
    <p
      style={
        Object {
          "lineHeight": "1.8",
          "textAlign": "justify",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </p>
  </div>
</div>
`;
